### ISSUE TITLE
object declared inside constructor

### ISSUE THREAD

**Issue by gavinking** [ceylon/ceylon-compiler#1967]
I have:

```ceylon
shared class X {
    shared new X() {
        object foo {}
    }
}
```

This results in:

```
compiler bug: unhandled declaration my.ceylononly::X.X with type Constructor at com.redhat.ceylon.compiler.java.codegen.AbstractTransformer.makeReifiedTypeArgumentResolved(AbstractTransformer.java:4705)
```

**Comment by tombentley** [comment]
Presumably fixed by the fix for #1992

**Comment by gavinking** [comment]
OK thanks:
